driver/event_table.c

```diff
@@ -451,21 +451,21 @@ const struct ppm_event_info g_event_info[] = {
 	[PPME_SYSCALL_EVENTFD2_E] = {"eventfd2", EC_IPC | EC_SYSCALL, EF_CREATES_FD | EF_MODIFIES_STATE, 1, {{"initval", PT_UINT64, PF_DEC} } },
 	[PPME_SYSCALL_EVENTFD2_X] = {"eventfd2", EC_IPC | EC_SYSCALL, EF_CREATES_FD | EF_MODIFIES_STATE, 2, {{"res", PT_FD, PF_DEC}, {"flags", PT_FLAGS16, PF_HEX, file_flags} } },
 	[PPME_SYSCALL_SIGNALFD4_E] = {"signalfd4", EC_SIGNAL | EC_SYSCALL, EF_CREATES_FD | EF_MODIFIES_STATE, 2, {{"fd", PT_FD, PF_DEC}, {"mask", PT_UINT32, PF_HEX}}},
-	[PPME_SYSCALL_SIGNALFD4_X] = {"signalfd4", EC_SIGNAL | EC_SYSCALL, EF_CREATES_FD | EF_MODIFIES_STATE, 2, {{"res", PT_FD, PF_DEC},  {"flags", PT_FLAGS16, PF_HEX}}},
+	[PPME_SYSCALL_SIGNALFD4_X] = {"signalfd4", EC_SIGNAL | EC_SYSCALL, EF_CREATES_FD | EF_MODIFIES_STATE, 2, {{"res", PT_FD, PF_DEC},  {"flags", PT_FLAGS16, PF_HEX, file_flags}}},
```

Comment:
Mmh i don't think `signalfd4` takes a file_flags flags: https://linux.die.net/man/2/signalfd4?

Reply:
> Starting with Linux 2.6.27, the following values may be bitwise ORed in flags to change the behaviour of signalfd():
>
> SFD_NONBLOCK
>
> Set the O_NONBLOCK file status flag on the new open file description. Using this flag saves extra calls to [fcntl](https://linux.die.net/man/2/fcntl)(2) to achieve the same result.
> 
> SFD_CLOEXEC
>
> Set the close-on-exec (FD_CLOEXEC) flag on the new file descriptor. See the description of the O_CLOEXEC flag in [open](https://linux.die.net/man/2/open)(2) for reasons why this may be useful.

Reply:
Technically it does not but we pretend it does: https://github.com/falcosecurity/libs/blob/master/driver/ppm_flag_helpers.h#L444-L472

We follow the same pattern in a few other events too (eventfd2 for example, IIRC)

Reply:
Thanks for the clarification, LGTM then :)